### ISSUE TITLE
localhost/127.0.0.1:7000 is in use by another process

### ISSUE THREAD

**Issue by tinymoe** [Impetus/kundera#11]
I am running Cassandra (0.8.2) standalone on my Mac after following the getting started steps. I am also using the "twitter like application" code from getting started.

When I run the TwitterExample I get the following error:

ERROR [Thread-5] 2011-09-16 16:29:59,591 AbstractCassandraDaemon.java (line 223) Fatal configuration error
org.apache.cassandra.config.ConfigurationException: localhost/127.0.0.1:7000 is in use by another process.  Change listen_address:storage_port in cassandra.yaml to values that do not conflict with other services
    at org.apache.cassandra.net.MessagingService.getServerSocket(MessagingService.java:218)
    at org.apache.cassandra.net.MessagingService.listen(MessagingService.java:191)
    at org.apache.cassandra.service.StorageService.joinTokenRing(StorageService.java:432)
    at org.apache.cassandra.service.StorageService.initServer(StorageService.java:412)
    at org.apache.cassandra.service.AbstractCassandraDaemon.setup(AbstractCassandraDaemon.java:219)
    at org.apache.cassandra.service.AbstractCassandraDaemon.activate(AbstractCassandraDaemon.java:341)
    at lucandra.CassandraUtils$1.run(CassandraUtils.java:259)
    at java.lang.Thread.run(Thread.java:680)

Although I do as it suggests (change storage_port in cassandra.yaml), I just get an error on the next port.

Any help is appreciated?

**Comment by mevivs** [comment]
Hi,
Which version of kundera are you using? Is it kundera -2.0.3?
I suspect Kundera-2.0.4 is getting updated for 2.0.4 which is bit unstable.
Looking to your error trace, it looks like issue with port is in use. try doing netstat or check for any process running on that port.

Thanks,
Vivek

**Comment by tinymoe** [comment]
Thanks for your reply Vivek.

I am using Kundera 2.0.3 (kundera-2.0.3-jar-with-dependancies.jar).

I have checked the port and it is a Cassandra service running on it. If you look in cassandra.yaml, you can see where it is specified:

# TCP port, for commands and data

storage_port: 7000

It's when the following line of code in the Twitter example that it fails: "return conf.getEntityManager(persistenceUnitName);"

I am assuming Kundera or the underlying Pelops client is reading the cassandra.yaml and also trying to start something on the same port. But that's a guess on my part as I am still learning how it all works.

**Comment by mevivs** [comment]
Hi,
Could you please try this by adding

```
        System.setProperty("cassandra.join_ring", "false");
```

just before

conf.getEntityManager(persistenceUnitName);

**Comment by tinymoe** [comment]
Thanks for your help. I get a different error now:

11/09/19 07:45:18 INFO pelops.PelopsClient: Added Solandra specific schema
java.lang.RuntimeException: Unable to create thrift socket to localhost/127.0.0.1:9160
    at org.apache.cassandra.thrift.CassandraDaemon$ThriftServer.<init>(CassandraDaemon.java:109)
    at org.apache.cassandra.thrift.CassandraDaemon.startServer(CassandraDaemon.java:56)
    at org.apache.cassandra.service.AbstractCassandraDaemon.startRPCServer(AbstractCassandraDaemon.java:285)
    at org.apache.cassandra.service.AbstractCassandraDaemon.start(AbstractCassandraDaemon.java:257)
    at lucandra.CassandraDaemon.start(CassandraDaemon.java:35)
    at org.apache.cassandra.service.AbstractCassandraDaemon.activate(AbstractCassandraDaemon.java:354)
    at lucandra.CassandraUtils$1.run(CassandraUtils.java:259)
    at java.lang.Thread.run(Thread.java:680)
Caused by: org.apache.thrift.transport.TTransportException: Could not create ServerSocket on address localhost/127.0.0.1:9160.
    at org.apache.thrift.transport.TServerSocket.<init>(TServerSocket.java:93)
    at org.apache.thrift.transport.TServerSocket.<init>(TServerSocket.java:79)
    at org.apache.cassandra.thrift.TCustomServerSocket.<init>(TCustomServerSocket.java:59)
    at org.apache.cassandra.thrift.CassandraDaemon$ThriftServer.<init>(CassandraDaemon.java:102)

Cassandra is running and I can it listening on port 9160:

INFO 07:43:46,777 Binding thrift service to localhost/127.0.0.1:9160
INFO 07:43:46,781 Using TFastFramedTransport with a max frame size of 15728640 bytes.
INFO 07:43:46,783 Listening for thrift clients...

**Comment by mevivs** [comment]
This looks to me some config issue.

see if given below thread gives any clue:

http://cassandra-user-incubator-apache-org.3065146.n2.nabble.com/Cassandra-still-won-t-start-in-use-ports-block-it-td6163698.html

**Comment by mevivs** [comment]
Are you able to connect to cassandra via cassandra-cli?

**Comment by tinymoe** [comment]
Yes, I can successfully connect via cassandra-cli and execute commands.

**Comment by tinymoe** [comment]
With regards to the thread you provided, no clues, sorry. Cassandra starts up cleanly and binds to the port successfully. The only time I see an issue is when connecting from the Java client.

**Comment by mevivs** [comment]
Okay..Could you please try 1 more command 

CassandraUtils.setStartup();

before 
conf.getEntityManager(persistenceUnitName);

I am wondering why are you getting this issue. which clearly is related to connecting to cassandra.

**Comment by mevivs** [comment]
Any luck?

**Comment by tinymoe** [comment]
Yes and no :)

I am connecting now. But, get a new error:

/09/19 09:35:37 WARN pelops.Connection: Failed to set keyspace on client.  See cause for details...
InvalidRequestException(why:Keyspace KunderaExamples does not exist)
    at org.apache.cassandra.thrift.Cassandra$set_keyspace_result.read(Cassandra.java:5302)
    at org.apache.cassandra.thrift.Cassandra$Client.recv_set_keyspace(Cassandra.java:481)
    at org.apache.cassandra.thrift.Cassandra$Client.set_keyspace(Cassandra.java:456)
    at org.scale7.cassandra.pelops.Connection.open(Connection.java:110)
    at org.scale7.cassandra.pelops.pool.CommonsBackedPool$ConnectionFactory.makeObject(CommonsBackedPool.java:891)
    at org.apache.commons.pool.impl.GenericKeyedObjectPool.addObject(GenericKeyedObjectPool.java:1685)
    at org.apache.commons.pool.impl.GenericKeyedObjectPool.ensureMinIdle(GenericKeyedObjectPool.java:2058)
    at org.apache.commons.pool.impl.GenericKeyedObjectPool.preparePool(GenericKeyedObjectPool.java:1722)
    at org.scale7.cassandra.pelops.pool.CommonsBackedPool.addNode(CommonsBackedPool.java:423)
    at org.scale7.cassandra.pelops.pool.CommonsBackedPool.<init>(CommonsBackedPool.java:137)
    at org.scale7.cassandra.pelops.pool.CommonsBackedPool.<init>(CommonsBackedPool.java:88)
    at org.scale7.cassandra.pelops.pool.CommonsBackedPool.<init>(CommonsBackedPool.java:76)
    at org.scale7.cassandra.pelops.Pelops.addPool(Pelops.java:48)
    at com.impetus.kundera.cassandra.client.pelops.PelopsClient.configurePool(PelopsClient.java:691)
    at com.impetus.kundera.cassandra.client.pelops.PelopsClient.writeData(PelopsClient.java:170)
    at com.impetus.kundera.cassandra.ColumnFamilyDataAccessor.write(ColumnFamilyDataAccessor.java:100)
    at com.impetus.kundera.db.DataManager.persist(DataManager.java:79)
    at com.impetus.kundera.ejb.EntityManagerImpl.persist(EntityManagerImpl.java:346)
    at com.impetus.kundera.examples.twitter.dao.TwitterService.addUser(TwitterService.java:56)
    at com.impetus.kundera.examples.twitter.dao.TwitterExample.main(TwitterExample.java:15)

Do I need to create the schema in Cassandra before I can use the client, or is it created automatically by virtue of the JPA annotations?

**Comment by mevivs** [comment]
Gosh...it is an issue with sample code then.. I will raise an issue to fix this in code base :)

Could you please use cassandra cli to  "create keyspace KunderaExamples" ?

**Comment by mevivs** [comment]
ddl_auto_create is still an open issue. Ideally TestSuite should have created this.

**Comment by tinymoe** [comment]
We are definitely making progress.

I ran "create keyspace KunderaExamples;"

However, I then got:

11/09/19 09:49:11 INFO pelops.Pelops: Pelops adds new pool Main
org.scale7.cassandra.pelops.exceptions.InvalidRequestException
at org.scale7.cassandra.pelops.exceptions.IExceptionTranslator$ExceptionTranslator.translate(IExceptionTranslator.java:47)
at org.scale7.cassandra.pelops.Operand.tryOperation(Operand.java:109)
at org.scale7.cassandra.pelops.Mutator.execute(Mutator.java:72)
at com.impetus.kundera.cassandra.client.pelops.PelopsClient.writeData(PelopsClient.java:192)
at com.impetus.kundera.cassandra.ColumnFamilyDataAccessor.write(ColumnFamilyDataAccessor.java:100)
at com.impetus.kundera.db.DataManager.persist(DataManager.java:79)
at com.impetus.kundera.ejb.EntityManagerImpl.persist(EntityManagerImpl.java:346)
at com.impetus.kundera.examples.twitter.dao.TwitterService.addUser(TwitterService.java:56)
at com.impetus.kundera.examples.twitter.dao.TwitterExample.main(TwitterExample.java:15)
Caused by: InvalidRequestException(why:unconfigured columnfamily users)
at org.apache.cassandra.thrift.Cassandra$batch_mutate_result.read(Cassandra.java:19045)
at org.apache.cassandra.thrift.Cassandra$Client.recv_batch_mutate(Cassandra.java:1035)
at org.apache.cassandra.thrift.Cassandra$Client.batch_mutate(Cassandra.java:1009)
at org.scale7.cassandra.pelops.Mutator$1.execute(Mutator.java:67)
at org.scale7.cassandra.pelops.Mutator$1.execute(Mutator.java:63)
at org.scale7.cassandra.pelops.Operand.tryOperation(Operand.java:82)

So I ran: "create column family users with comparator = UTF8Type;" (not sure if that's correct?)

I then got the following error, but I am not sure what command to execute now?

1/09/19 09:51:28 INFO pelops.Pelops: Pelops adds new pool Main
org.scale7.cassandra.pelops.exceptions.InvalidRequestException
    at org.scale7.cassandra.pelops.exceptions.IExceptionTranslator$ExceptionTranslator.translate(IExceptionTranslator.java:47)
    at org.scale7.cassandra.pelops.Operand.tryOperation(Operand.java:109)
    at org.scale7.cassandra.pelops.Mutator.execute(Mutator.java:72)
    at com.impetus.kundera.cassandra.client.pelops.PelopsClient.writeData(PelopsClient.java:192)
    at com.impetus.kundera.cassandra.ColumnFamilyDataAccessor.write(ColumnFamilyDataAccessor.java:100)
    at com.impetus.kundera.db.DataManager.persist(DataManager.java:79)
    at com.impetus.kundera.ejb.EntityManagerImpl.persist(EntityManagerImpl.java:346)
    at com.impetus.kundera.examples.twitter.dao.TwitterService.addUser(TwitterService.java:56)
    at com.impetus.kundera.examples.twitter.dao.TwitterExample.main(TwitterExample.java:15)
Caused by: InvalidRequestException(why:supercolumn parameter is invalid for standard CF users)
    at org.apache.cassandra.thrift.Cassandra$batch_mutate_result.read(Cassandra.java:19045)
    at org.apache.cassandra.thrift.Cassandra$Client.recv_batch_mutate(Cassandra.java:1035)
    at org.apache.cassandra.thrift.Cassandra$Client.batch_mutate(Cassandra.java:1009)
    at org.scale7.cassandra.pelops.Mutator$1.execute(Mutator.java:67)
    at org.scale7.cassandra.pelops.Mutator$1.execute(Mutator.java:63)
    at org.scale7.cassandra.pelops.Operand.tryOperation(Operand.java:82)

**Comment by mevivs** [comment]
Try this..

create column family users with column_type = ‘Super’ and comparator = ‘UTF8Type’ and subcomparator = UTF8Type;

**Comment by tinymoe** [comment]
OK, I tried that, but I got another error...

java.lang.NullPointerException
    at org.apache.cassandra.db.ColumnFamily.<init>(ColumnFamily.java:82)
    at org.apache.cassandra.db.ColumnFamily.create(ColumnFamily.java:70)
    at org.apache.cassandra.db.ColumnFamily.create(ColumnFamily.java:65)
    at org.apache.cassandra.db.RowMutation.add(RowMutation.java:150)
    at org.apache.cassandra.db.RowMutation.add(RowMutation.java:170)
    at lucandra.CassandraUtils.addMutations(CassandraUtils.java:525)
    at lucandra.IndexWriter.addDocument(IndexWriter.java:250)
    at com.impetus.kundera.index.KunderaIndexer.indexDocument(KunderaIndexer.java:411)
    at com.impetus.kundera.index.KunderaIndexer.onPersist(KunderaIndexer.java:318)
    at com.impetus.kundera.index.KunderaIndexer.indexSuperColumn(KunderaIndexer.java:275)
    at com.impetus.kundera.index.KunderaIndexer.performIndexing(KunderaIndexer.java:223)
    at com.impetus.kundera.index.KunderaIndexer.index(KunderaIndexer.java:174)
    at com.impetus.kundera.index.IndexManager.write(IndexManager.java:119)
    at com.impetus.kundera.ejb.EntityManagerImpl.persist(EntityManagerImpl.java:347)
    at com.impetus.kundera.examples.twitter.dao.TwitterService.addUser(TwitterService.java:56)
    at com.impetus.kundera.examples.twitter.dao.TwitterExample.main(TwitterExample.java:15)
Exception in thread "main" javax.persistence.PersistenceException: java.lang.NullPointerException
    at com.impetus.kundera.ejb.EntityManagerImpl.persist(EntityManagerImpl.java:356)
    at com.impetus.kundera.examples.twitter.dao.TwitterService.addUser(TwitterService.java:56)
    at com.impetus.kundera.examples.twitter.dao.TwitterExample.main(TwitterExample.java:15)
Caused by: java.lang.NullPointerException
    at org.apache.cassandra.db.ColumnFamily.<init>(ColumnFamily.java:82)
    at org.apache.cassandra.db.ColumnFamily.create(ColumnFamily.java:70)
    at org.apache.cassandra.db.ColumnFamily.create(ColumnFamily.java:65)
    at org.apache.cassandra.db.RowMutation.add(RowMutation.java:150)
    at org.apache.cassandra.db.RowMutation.add(RowMutation.java:170)
    at lucandra.CassandraUtils.addMutations(CassandraUtils.java:525)
    at lucandra.IndexWriter.addDocument(IndexWriter.java:250)
    at com.impetus.kundera.index.KunderaIndexer.indexDocument(KunderaIndexer.java:411)
    at com.impetus.kundera.index.KunderaIndexer.onPersist(KunderaIndexer.java:318)
    at com.impetus.kundera.index.KunderaIndexer.indexSuperColumn(KunderaIndexer.java:275)
    at com.impetus.kundera.index.KunderaIndexer.performIndexing(KunderaIndexer.java:223)
    at com.impetus.kundera.index.KunderaIndexer.index(KunderaIndexer.java:174)
    at com.impetus.kundera.index.IndexManager.write(IndexManager.java:119)
    at com.impetus.kundera.ejb.EntityManagerImpl.persist(EntityManagerImpl.java:347)
    ... 2 more

**Comment by mevivs** [comment]
Can you see if there is any keyspace created by name of "L". ?

**Comment by mevivs** [comment]
Hi tinymoe,
I think there is some with configuration. Leave it to me.i will update you on this after discussion with team.
I can sense what the issue is.

**Comment by tinymoe** [comment]
Thanks for all your help. I am looking forward to getting the example working :)

**Comment by mevivs** [comment]
By tomorrow  i will update you on this to get this working. This is platform specific issue.
I hope you must have configured your setup as described  on 

https://github.com/impetus-opensource/Kundera/wiki/Setup-Kundera-with-standalone-Cassandra%3A

**Comment by tinymoe** [comment]
Yes I have followed the steps described in the above link.

**Comment by tinymoe** [comment]
Did you have any success resolving this issue?

Thanks.

**Comment by mevivs** [comment]
Apologies as did not get time to look into this.
Assigning somebody to look into this.

Sincerely,
Vivek

**Comment by mevivs** [comment]
Hi,
Sorry for reply after a long time. 

Please try with 2.0.4 release. All such issues are resolved now.

**Comment by xamry** [comment]
Fixed in version 2.0.4, closing.

**Comment by Kamaraju333** [comment]
Xamry , 

what fix was made to fix this issue ?
